app/templates/entry/client-prefetch.js

```diff
@@ -60,18 +63,25 @@ export function addPreFetchHooks (router<%= store ? ', store' : '' %>, publicPat
           m.path.indexOf('/:') > -1 // does it has params?
         ))
       })
-      .filter(m => m.c && typeof m.c.preFetch === 'function')
-      .map(m => m.c.preFetch)
+      .filter(m => {
```

Comment:
Why not simplify this to be
```ts
      .filter(m => {
        // Class components return the component options (and the preFetch hook) inside __c property
        return (m.c && typeof m.c.preFetch === 'function') || (m.c && m.c.__c && typeof m.c.__c.preFetch === 'function')
      })
```

Reply:
Probably readability, as it's easier for the eye to understand those are branches for different use-case.
But the comment I asked to add solves the same problem, we can further simplify it as

```ts
// Class components return the component options (and the preFetch hook) inside __c property
.filter(m => (m.c && typeof m.c.preFetch === 'function') || (m.c && m.c.__c && typeof m.c.__c.preFetch === 'function'))
.map(m => m.c.__c ? m.c.__c.preFetch : m.c.preFetch)
``` 

Since if aren't required anymore

Reply:
Done!